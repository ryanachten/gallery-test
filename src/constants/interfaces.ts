export type Action = {
  type: string;
  payload?: any;
  error?: any;
};

export enum Size {
  ALL = "",
  XS = "XS",
  S = "S",
  M = "M",
  L = "L",
  XL = "XL"
}

export type Product = {
  index: number;
  isSale: boolean;
  isExclusive: boolean;
  price: string;
  productImage: string;
  productName: string;
  size: Size[];
};
