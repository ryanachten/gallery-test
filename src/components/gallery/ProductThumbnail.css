.ProductThumbnail {
  align-items: flex-start;
  border: 1px solid #e1e1e1;
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  padding: 1.125em 0.68em;
  width: 320px;
}

.ProductThumbnail--img {
  width: 100%;
}

.ProductThumbnail--meta {
  align-items: flex-end;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  width: 100%;
}

.ProductThumbnail--flag {
  display: flex;
  justify-content: flex-start;
  width: 100%;
}

.ProductThumbnail--flag-sale {
  background: #cc3333;
  color: white;
  margin-bottom: 0.31em;
  padding: 0.75em 1em;
}

.ProductThumbnail--flag-exclusive {
  background: #009900;
  color: white;
  margin-bottom: 0.31em;
  padding: 0.75em 1em;
}

.ProductThumbnail--name {
  font-size: 1.5em;
  font-weight: bold;
  margin: 0;
  max-width: 50%;
}

.ProductThumbnail--price {
  font-size: 2.25em;
  font-weight: bold;
  margin: 0;
}
