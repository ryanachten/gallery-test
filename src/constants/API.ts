export const API_URL = "https://api.jsonbin.io/b/5cae9a54fb42337645ebcad3";
