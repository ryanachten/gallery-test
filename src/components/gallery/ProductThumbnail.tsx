import React from "react";
import { Product } from "../../constants/interfaces";
import "./ProductThumbnail.css";

type Props = {
  product: Product;
};

const ProductThumbnail = (props: Props) => {
  const {
    index,
    isExclusive,
    isSale,
    price,
    productImage,
    productName
  } = props.product;
  return (
    <article className="ProductThumbnail" key={index}>
      <img
        className="ProductThumbnail__img"
        src={`products/${productImage}`}
        alt={productName}
      />
      {isSale && <span className="ProductThumbnail__flag--sale">Sale</span>}
      {isExclusive && (
        <span className="ProductThumbnail__flag--exclusive">Exclusive</span>
      )}
      <section className="ProductThumbnail__meta">
        <p className="ProductThumbnail__name">{productName}</p>
        <p className="ProductThumbnail__price">{price}</p>
      </section>
    </article>
  );
};

export default ProductThumbnail;
