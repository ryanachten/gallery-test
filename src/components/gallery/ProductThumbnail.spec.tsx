import React from "react";
import renderer from "react-test-renderer";
import ProductThumbnail from "./ProductThumbnail";
import { Product, Size } from "../../constants/interfaces";

describe("Component: ProductThumbnail", () => {
  const productTemplate: Product = {
    index: 0,
    isSale: true,
    isExclusive: false,
    price: "$18.88",
    productImage: "product-1.jpg",
    productName: "Striped shirt",
    size: [Size.XS, Size.S, Size.L, Size.XL]
  };

  const initProps = {
    getProducts: jest.fn(),
    product: productTemplate
  };

  it("renders correctly", () => {
    const tree = renderer.create(<ProductThumbnail {...initProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
