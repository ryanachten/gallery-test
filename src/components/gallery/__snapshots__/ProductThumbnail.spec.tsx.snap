// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductThumbnail renders correctly 1`] = `
<article
  className="ProductThumbnail"
>
  <img
    alt="Striped shirt"
    className="ProductThumbnail--img"
    src="products/product-1.jpg"
  />
  <span
    className="ProductThumbnail--flag-sale"
  >
    Sale
  </span>
  <section
    className="ProductThumbnail--meta"
  >
    <p
      className="ProductThumbnail--name"
    >
      Striped shirt
    </p>
    <p
      className="ProductThumbnail--price"
    >
      $18.88
    </p>
  </section>
</article>
`;
